### ISSUE TITLE
Inviting People to Slack

### ISSUE THREAD

**Issue by dselman** [accordproject/technical-steering-committee#18]
# Discussion 🗣

Slack allows admins to create an open sign-up link, however these are limited to 2,000 sign-ups, and must be recreated when the limit it used. They may also have an expiration date.

## Context

Slack sign-up link is irregularly broken.

## Detailed Description

https://slack.com/intl/en-gb/help/articles/201330256-Invite-new-members-to-your-workspace

**Comment by dselman** [comment]
Invite via APIs perhaps?
https://intoli.com/blog/make-a-public-slack-community/

**Comment by martinhalford** [comment]
FYI... There's a chance Accord might be able to get a free upgrade to the Slack Enterprise Grid plan. 

Reached out to Slack and received the following response.  

> Sam (Slack) Jan 29, 2021, 3:41 PM PST Hi Martin, Thanks for reaching out with your question! Our Enterprise Grid plan for nonprofits requires a minimum member requirement of 1,000 members. You can read all the details about the program, including its eligibility requirements, in our Help Center: https://get.slack.help/hc/en-us/articles/204368833-Slack-for-Nonprofits-programLooking forward to hearing back! Kindly, Sam

According to the Slack Help Centre link:
> We welcome applications from organisations that hold a valid charitable status with either the IRS (in the United States), a local tax service/charity commission or a local TechSoup Global partner.

I'm assuming this means we need to get our 501(c)3 registration sorted before we can apply maybe?

Once we think we qualify, someone with Accord Project Slack Admin privileges we can apply via [https://slack.com/nonprofit](https://slack.com/nonprofit).

**Comment by martinhalford** [comment]
FYI... Further discussions with Slack.  Apparently, although we have 1000's of registered members in the Accord Project Slack Channel, we only have 48 active members.  Therefore, we don't qualify for (or need) the Enterprise Grid licence as the Grid licence is only for organisations with 1000+ active members.

> Sam (Slack) Feb 1, 2021, 12:12 PM PST 
Hi Martin,
>
>Yes! You have thousands of members added to the workspace, but only 48 of those members are active. On our end, implementing the Grid plan isn't possible unless 1,000+ members are actively using the platform.
>
>I'd recommend applying for the Slack for Nonprofits program and choosing the Standard plan. Once approved, you will receive a complimentary upgrade to the Standard plan for your whole workspace. You can use this time to get your members set up and active. Once you reach 1,000+ active members, feel free to touch base with us, and I'd be happy to get you in touch with a member of the Sales team that can assist with switching to Grid.
>
>Let me know if this clarifies things!
>
>Kindly, 
>Sam

@irmerk @jeromesimeon

**Comment by martinhalford** [comment]
However.... the good news is that, if we can prove our Nonprofit status then we **DO** qualify for a free upgrade to the **Standard plan**.  We just need our 501(c)3 status confirmed.

>Sam (Slack) Feb 1, 2021, 1:48 PM PST 
Great! 
Happy to help. Our partners at TechSoup will be reviewing your application, but I can confirm they will be looking for certification of your nonprofit status, so I'd recommend sending that all in on one go when you've got confirmation.
Hope this helps!
Sam

@irmerk @jeromesimeon

**Comment by jeromesimeon** [comment]
> However.... the good news is that, if we can prove our Nonprofit status then we **DO** qualify for a free upgrade to the **Standard plan**. We just need our 501(c)3 status confirmed.
> 
> > Sam (Slack) Feb 1, 2021, 1:48 PM PST 
> > Great!
> > Happy to help. Our partners at TechSoup will be reviewing your application, but I can confirm they will be looking for certification of your nonprofit status, so I'd recommend sending that all in on one go when you've got confirmation.
> > Hope this helps!
> > Sam
> 
> @irmerk @jeromesimeon

That is good news (also a useful reality check).

I would bounce that question back to the Linux foundation?

**Comment by martinhalford** [comment]
@jeromesimeon - seems we definitely need some sort of certification of our status as a Nonprofit.  

If the Linux Foundation can provide that, then great.  Suspect a 501(c)3 might carry more gravitas.  

I think @anton1s is looking at the 501(c)3 as part of [#Issue 19](https://github.com/accordproject/technical-steering-committee/issues/19).  

Maybe we can solve both problems at once?

**Comment by dselman** [comment]
I've confirmed that invites are currently not working, due to the infamous `invite_limit_reached` error - which is (apparently) based on the ratio of invites sent to invites accepted.

I installed Inviter Bot to test: https://invite.playplay.io/invite?team_id=T5NC5T9EX

**Comment by dselman** [comment]
This is also an interesting thread on some alternatives to Slack for open communities: https://wiki.p2pfoundation.net/Slack_Alternatives

**Comment by dselman** [comment]
```
THE LINUX FOUNDATION is headquartered in SAN FRANCISCO, CA, and is a 501(c)(6) organization. EIN: 46-0503801.  Donations may or may not be tax-deductible. The IRS NTEE classification code is S41, Promotion of Business within the Community Improvement, Capacity Building category. The IRS ruling year for tax exemption was 2003.
```

https://www.charitynavigator.org/ein/460503801

**Comment by martinhalford** [comment]
@dselman - If we think we qualify already for Slack's Charity Discount then someone with Accord Project Slack Channel Admin privileges can apply today.

https://slack.com/intl/en-gb/help/articles/204368833-Apply-for-the-Slack-for-Charities-discount

Getting off of the Free Tier and onto the Standard Plan should, maybe, remove this limitation?

**Comment by dselman** [comment]
I will give that a try.

**Comment by martinhalford** [comment]
Follow up with Slack. See if Slack will help. Otherwise, look to Discord or others.

**Comment by martinhalford** [comment]
@anton1s to investigate a bit further. @irmerk has reminder set to reset Slack.

**Comment by martinhalford** [comment]
From TSC Meeting 25 March 2021:
```
- No communication from Slack about potential upgrade for free due to AP's not-for-profit nature.
- Antonis volunteered to make one last attempt to reach out to Slack.
- Main feature AP needs is to not have a user cap in order to make it easier for people to join.
- Currently AP can only invite members via a "Group" invite which expires every 30 days,
- Otherwise, may be worth considering Discord.
- Will need to consider effort and process of moving from Slack to Discord.
- Agreed to postpone decision until next TSC meeting.
- If AP was upgraded to "Standard" plan for free, then AP would be limit to 250 Active Users. Subsequent users would be at 85% of retail cost.  
- Cost would be $1.95 per users above the 250 active users for standard "free" tier.
- Likely that the "Group" expiration limit would still exist even after free upgrade due to security reasons. 
- **ACTION** 
	- Antonis to see if he can find out as much information as possible via his contacts at Slack.
- **ACTION** 
	- Jolene to look into Discord limitations (if any).
```

**Comment by martinhalford** [comment]
From TSC Meeting 29 Apr 2021:

- No further updates since previous meeting.
- Need to still determine if monthly Invite link expiration still occurs under upgraded "Standard" plan. Suspect answer is "yes".
- Another Slack replacement option to consider is [https://orbit.love/](https://orbit.love/).
- Defer decision to next TSC meeting pending further info from discussions with Slack and investigations into Orbit.

**Comment by anweiss** [comment]
hey all! new to the community and trying to join the Slack workspace, but it seems the link is down again based on what's described by this issue. Is there a new, temporary signup link we can use at least until this issue is resolved? thanks!

**Comment by irmerk** [comment]
@jeromesimeon you had mentioned this on yesterdays working group call. I didn't catch if this is resolved for the time being?

**Comment by martinhalford** [comment]
From TSC Meeting: Thursday 27 May, 2021:

- No possibility of receiving free upgrade from Slack. All avenues exhausted.
- No further details regarding Orbit.
- Need a "champion" to do the work of migrating from Slack to Discord or Orbit or some other platform.
- At the moment, with GSoC etc. taking time, decision to postpone decision until a future meeting.

**Comment by martinhalford** [comment]
From TSC Meeting: Thursday 24 June, 2021:
- Issue remains unresolved with clunky work-around still in place.
- No "champion" to implement replacement at this time.
- To be reviewed at next month's TSC meeting.

**Comment by martinhalford** [comment]
@jeromesimeon @dselman @irmerk - I've been doing some research on Slack alternatives, and how we might migrate the Accord Project community from Slack to, say, Discord.  

Helpfully, we are not the first to do this!

Orbit.love (which I thought were a competitor to Discord) migrated their community from Slack to Discord a short while back and have posted their findings as follows:

- A really good "pro's and con's" summary. [Slack vs Discord vs Discourse: The best tool for your community](https://orbit.love/blog/slack-vs-discord-vs-discourse)

- The process they following when migrating from one platform to the next. [Migrating Our Community from Slack to Discord: Tips & Lessons Learned](https://orbit.love/blog/how-to-migrate-a-community-from-slack-to-discord)

- The document Orbit shared with their community to encourage engagement and maintain transparency. [Proposal: migrating Orbit One from Slack to Discord](https://docs.google.com/document/d/1PFHhZyyYmiIHNufYpS6fX7gUMEJVo3FM0Y0RZa7DuDo/edit)

- We'll need to understand Roles when setting up a Discord "Server"[Role Management in Discord](https://support.discord.com/hc/en-us/articles/214836687-Role-Management-101)

![Comparing Slack, Discord, and Discourse](https://cdn.sanity.io/images/cad8jutx/production/aff197588f79ad746695aefe4c436e90b1aa904b-1220x3044.png)

**Comment by pkel** [comment]
Another direct competitor to Slack is https://mattermost.com/. It comes with a free, self-hosted plan. I've never used it seriously, thus I cannot give a strong recommendation.

Apparently, they also thought about migration from Slack: https://docs.mattermost.com/onboard/migrating-to-mattermost.html#migrating-from-slack

**Comment by martinhalford** [comment]
Kushal also recommended https://zulip.com in the TechWG Zoom chat today.

**Comment by martinhalford** [comment]
@jeromesimeon @irmerk - Following on from our discussion during the Tech WG call about Slack, someone suggested an email blast to all Slack Members to advise them of the proposal.  Apparently an Accord Project Slack Admin should be able to generate a .csv file of all members which includes their email addresses.

<img width="1587" alt="image" src="https://user-images.githubusercontent.com/970336/128293999-ca4d96eb-6e56-4431-a8ac-604340ac1562.png">

**Comment by martinhalford** [comment]
From TSC Meeting 36 Aug 2021:
- Currently trialling Discord (for about a week at time of TSC meeting).
- General consensus is positive.
- Able to migrate GitHub integrations from Slack to Discord.
- Inviting members is much simpler and not subject to links expiring.
- Similar user experience in Discord compared to Slack with equivalent channels replicated in Discord.
- **ACTIONS** 
	- Formally migrate from Accord Project to Discord.
	- Provide two weeks notice to participants.
	- Archive Slack channels after notice period.
	- Replace links to Slack with links to Discord.
	- Decided **not** to send email blast, due to effort, cost and lack of permission to send unsolicited emails.

**Comment by martinhalford** [comment]
Migration to Discord complete. Yay!!

**Comment by irmerk** [comment]
This is leftover that we need to do: https://github.com/accordproject/techdocs/issues/388
